text/0015-plugin-migrations.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: removable_plugin_migrations
+- Start Date: 2016-09-13
+- RFC PR:
+- Issue:
+
+# Summary
+[summary]: #summary
+
+A suggestion how to improve the ability to migrate foreman's database
+when installing and uninstalling a plugin.
+
+# Motivation
+[motivation]: #motivation
+
+One of the biggest pain points of uninstalling a plugin is the database
+leftovers. While the code could be easily removed, database objects that were
+added by the plugin remain.
+
+# Detailed design
+[design]: #detailed-design
+
+I want to base my solution on the fact that rails migrations have a notion of
+[scoping](http://edgeguides.rubyonrails.org/engines.html#engine-setup).
+I suggest creating naming convention for migrations added by a plugin: the name
+of each file should look like this:
+
+*timestamp*_*migration_name*.*PLUGIN_NAME*.rb
```

Comment:
Overall I agree, but can you be very explicit about the `PLUGIN_NAME` here? In the past, we had some plugins that were incorrectly named (e.g. discovery vs foreman_discovery inconsistency in constants, plugin name and some namespaces). I think we should state what this tag should match to (is it plugins main module name? camelcase or underscore with downcase?)

I think I can assume `foreman_plugin_name` here, but let's be explicit.